src/System.Drawing.Common/src/System/Drawing/Printing/PrintingServices.Unix.cs

```diff
@@ -55,30 +56,43 @@ internal static class PrintingServices
         private static Hashtable installed_printers;
         private static string default_printer = string.Empty;
 
+        private static volatile bool populatedPrinters = false;
+
         #endregion
 
         #region Constructor
 
         static PrintingServices()
         {
-            installed_printers = new Hashtable();
             CheckCupsInstalled();
         }
 
         #endregion
 
         #region Properties
 
+        private static Hashtable ReadPrinters()
+        {
+            var printers = new Hashtable();
+            printers = LoadPrinters(printers);
+            return printers;
+        }
```

Comment:
Is LoadPrinters called from anywhere other than this method now?  Can we instead just make LoadPrinters create the Hashtable itself and return it, rather than taking one as an argument?  Then this ReadPrinters method isn't needed at all, and the uses of ReadPrinters can be replaced by LoadPrinters.